src/query_plan/operation.rs

```diff
@@ -72,23 +86,23 @@ pub(crate) enum NormalizedSelectionKey {
         // directives applied on the field
         directives: Arc<DirectiveList>,
         // unique label/counter used to distinguish fields that cannot be merged
-        label: i32,
+        label: usize,
```

Comment:
Why are these `usize` rather than `SelectionId`?

Reply:
The gist is it can be one of two possibilities: `0` to indicate that the corresponding selection doesn't use defer, or the selection ID if the selection does use defer. (We could use enums here instead, where one variant corresponds to what `0` currently means, and the other variant stores a selection ID.)

Reply:
updated to `Option<SelectionId>`